src/reducers/tokenList.ts

```diff
@@ -1,20 +1,39 @@
 import { handleActions } from 'redux-actions'
 
 import { setDefaultTokenList, setCustomTokenList } from 'actions'
+import { setTokenListType } from 'containers/TokenUpload'
 
 export default handleActions(
   {
     [setDefaultTokenList.toString()]: (state: any, action: any) => ({
       ...state,
       defaultTokenList: action.payload.defaultTokenList,
+      combinedTokenList: [
+        ...state.customTokenList,
+        ...action.payload.defaultTokenList,
+      ],
     }),
     [setCustomTokenList.toString()]: (state: any, action: any) => ({
       ...state,
       customTokenList: action.payload.customTokenList,
+      combinedTokenList: [
+        ...state.defaultTokenList,
+        ...state.customTokenList,
+        ...action.payload.customTokenList,
```

Comment:
this will always add to the current `combinedTokenList`. Try uploading the same file twice and see the duplicates appear.
Removing `...state.customTokenList` will fix that.